src/components/generic/MenuFilter.tsx

```diff
@@ -0,0 +1,164 @@
+        }
+        if (selectedValues.length !== originalSelectedValues.length
+            || selectedValues.slice(0).sort().join() !== originalSelectedValues.slice(0).sort().join()) {
+            return true;
+        }
+        return false;
+    }
+
+    onChange(newValue, checked, event) {
+        var { selectedValues } = this.state;
+        let newSelectedValues = selectedValues.slice(0);
+
+        const idx = selectedValues.findIndex((x) => x === newValue);
+        if (idx === -1 && checked) {
+            newSelectedValues.push(newValue);
+        } else if (idx > -1 && !checked) {
+            newSelectedValues.splice(idx, 1);
+        } else {
+            console.warn("Unexpected checked filter state:", newValue, checked);
+        }
+        this.setState({ selectedValues: newSelectedValues });
+    }
+
+    render() {
+        var { title, subtitle, icon } = this.props;
+        var { selectedValues, values, overlay } = this.state;
+        values = values || [];
+
+        const button = title === "" ?
+            <Button icon tooltipLabel={subtitle} onClick={this.toggleOverlay}>{icon}</Button>
+            : <Button flat label={title} onClick={this.toggleOverlay}>{icon}</Button>;
+
+        let listItems = values.map((value, idx) => {
+            return (
+                <ListItemControl
+                    key={idx + title}
+                    primaryAction={(
+                        <Checkbox
+                            id={idx + value}
+                            name={value}
+                            label={value}
+                            onChange={this.onChange.bind(null, value)}
+                            checked={selectedValues.find((x) => x === value) !== undefined}
+                        />
+                    )}
+                />
+            );
+        })
+
+        if (values.length > 1) {
+            const selectAll = this.props.selectAll;
+            const selectNone = this.props.selectNone;
+            listItems.push(<ListItem key="all" primaryText={selectAll} onClick={this.selectAll} rightIcon={<FontIcon>done_all</FontIcon>} />);
+            listItems.push(<ListItem key="none" primaryText={selectNone} onClick={this.selectNone} rightIcon={<FontIcon disabled>check_box_outline_blank</FontIcon>} />);
+        }
+
+        return (
+            <div className="filters">
+                {button}
+
+                <div style={styles.container} hidden={!overlay}>
+                    <List className="md-paper md-paper--1" style={styles.list}>
+                        {listItems}
+                    </List>
+                </div>
+
+                <Portal visible={overlay}>
+                    <AccessibleFakeButton
+                        className="md-overlay"
+                        onClick={this.hideOverlay}
+                        style={styles.overlay}
+                    />
+                </Portal>
+            </div>
+        );
+    }
+
+    selectAll() {
```

Comment:
Try keeping private methods before render (Not sure if this is best practice, but we should make the code consistent)